internal/pkg/source/source.go

```diff
@@ -266,3 +287,13 @@ func isProducedBySanitizer(v *ssa.Alloc, conf classifier) bool {
 	}
 	return false
 }
+
+func indexInBlock(target ssa.Instruction) int {
+	for i, instr := range target.Block().Instrs {
+		if instr == target {
+			return i
+		}
+	}
+	// this cannot happen
+	return -1
```

Comment:
Those are betting words.

If there is a contract of correct use, put it in a docstring.

I'm mildly opposed to sentinel values.  Consider using the return pair `(int, ok)` or `(int, error)` instead.

Reply:
Hitting that `return -1` would mean that we've encountered a pretty severe bug in the SSA package, namely that an instruction isn't actually in the block it claims to be in.

The `ssa` package actually has a sanity checker that verifies that this does not happen (`ssa/sanity.go:320`), but it doesn't look like it gets called when using `buildssa`, so I modified the code to use the `(int, bool)` form.